### ISSUE TITLE
Crash in VirtualBox, VMWare

### ISSUE THREAD

**Issue by Sparkpin** [Sparkpin/spinel#9]
Attempts to kernel panic when mapping a page, but triple faults:
```
00:00:03.774035 4000 MiB memory available.                                                      
00:00:03.774044 Got scan code 0x9C                                                              
00:00:03.774053 Page fault at 0xC0400000                                                        
00:00:03.774062 Page fault at 0xFFF01000                                                        
00:00:03.774071 panic: Page fault                                                               
00:00:03.774080 CR2 0xFFF01000    Error code 0x                                                 
00:00:03.774089                                          
```

```
00:00:03.771180 Guest CPUM (VCPU 0) state: 
00:00:03.771184 eax=00006eaa ebx=c010deaa ecx=ffff9155 edx=00000000 esi=00000000 edi=c010deaa
00:00:03.771186 eip=c0103130 esp=c010de6c ebp=c010df08 iopl=0         nv up di pl zr na po cy
00:00:03.771188 cs={0008 base=0000000000000000 limit=ffffffff flags=0000c09b} dr0=00000000 dr1=00000000
00:00:03.771190 ds={0010 base=0000000000000000 limit=ffffffff flags=0000c093} dr2=00000000 dr3=00000000
00:00:03.771191 es={0010 base=0000000000000000 limit=ffffffff flags=0000c093} dr4=00000000 dr5=00000000
00:00:03.771193 fs={0010 base=0000000000000000 limit=ffffffff flags=0000c093} dr6=ffff0ff0 dr7=00000400
00:00:03.771195 gs={0010 base=0000000000000000 limit=ffffffff flags=0000c093} cr0=80000011 cr2=c0106fff
00:00:03.771196 ss={0010 base=0000000000000000 limit=ffffffff flags=0000c093} cr3=00108000 cr4=00000000
00:00:03.771197 gdtr=00000000c010e060:002f  idtr=00000000c010e120:07ff  eflags=00200007
```

**Comment by Sparkpin** [comment]
May be related to #10

**Comment by Sparkpin** [comment]
af9a2f2 fixed the triple fault. Now the page fault can show:

```
Spinel 0.1.0 2020-03-12 10:34pm CDT d93da0e on i386 PC compatible
Kernel size is 58 KiB.
4000 MiB memory available.
Got scan code 0x9C
Page fault at 0xC0400000
Page fault at 0xFFF01000
panic: Page fault
CR2 0xFFF01000    Error code 0x0

Page table entry 0x0    Page directory entry 0x108023
```

**Comment by Sparkpin** [comment]
```
Spinel 0.1.0 2020-04-10 12:56pm CDT 49487d2 on i386 PC compatible
Kernel size is 62 KiB.
4000 MiB memory available.
Page fault at 0xC0400000
Page fault at 0xFFF01000
panic: Page fault
CR2 0xFFF01000    Error code 0x0

Page table entry 0x0    Page directory entry 0x109023
Backtrace:
    0xC01027CC [panic + 60 in section .text]
    0xC0101467 [isrCommon + 7 in section .text]
    0xC0101467 [isrCommon + 7 in section .text]
    0xC0100296 [processMultibootInfo + 70 in section .text]
    0xC01001F5 [archBoot + 133 in section .text]
    0xC0100246 [_start.pagingDone + 19 in section .text]

**Comment by Sparkpin** [comment]
Fixed, presumably by the virtual memory manager overhaul (8497440) or by the invalid memory check fix (db3a2de)

![](https://pls.pet/u/Screenshot_20200502_023430.png)
